client/pk-console.c

```diff
@@ -637,30 +637,21 @@ pk_console_progress_cb (PkProgress *progress, PkProgressType type, gpointer data
 }
 
 static void
-pk_console_finished_cb (GObject *object, GAsyncResult *res, gpointer data)
+pk_console_process_results (PkResults *results, PkConsoleCtx *ctx, GError *err)
 {
 	const gchar *filename;
 	gboolean ret;
 	GPtrArray *array;
 	PkPackageSack *sack;
 	PkRestartEnum restart;
 	PkRoleEnum role;
-	PkConsoleCtx *ctx = (PkConsoleCtx *) data;
 	g_autoptr(GError) error = NULL;
 	g_autoptr(GFile) file = NULL;
 	g_autoptr(PkError) error_code = NULL;
-	g_autoptr(PkResults) results = NULL;
 
-	/* no more progress */
-	if (ctx->is_console) {
-		pk_progress_bar_end (ctx->progressbar);
-	} else {
-		/* TRANSLATORS: the results from the transaction */
-		g_print ("%s\n", _("Results:"));
-	}
+	if (err != NULL)
+		error = g_error_copy (err);
```

Comment:
Why are you copying the error here? Why not get rid of the local error and use the passed GError directly? (Ideally checking it for NULL before using it, just to be safe)

Reply:
Because of this below code in the function. I just wanted to make minimal changes.

```c
	/* write the sack to disk */
	if (role == PK_ROLE_ENUM_GET_PACKAGES && filename != NULL) {
		file = g_file_new_for_path (filename);
		ret = pk_package_sack_to_file (sack, file, &error);
		if (!ret) {
			g_print ("%s: %s\n", _("Fatal error"), error->message);
			ctx->retval = PK_EXIT_CODE_TRANSACTION_FAILED;
			goto out;
		}
	}
```

Reply:
Right, but please don't reuse a GError like that, that's a recipe for disaster usually.

Just make a local `g_autoptr(GError) error = NULL` for the function, and keep the `finish_error` that has been passed to the function intact and unchanged. For the error to work for `pk_package_sack_to_file`, it has to be NULL anyway.

Reply:
> (Ideally checking it for NULL before using it, just to be safe)

If we're planning to do that, we should do that in all places. Personally, we should just inspect the code and fix any programmer errors when `error` is NULL when it should not, rather than adding checks. So, I'm skipping the NULL check for now.